opslevel/resource_opslevel_trigger_definition.go

```diff
@@ -0,0 +1,161 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceTriggerDefinition() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a webhook action",
+		Create:      wrap(resourceTriggerDefinitionCreate),
+		Read:        wrap(resourceTriggerDefinitionRead),
+		Update:      wrap(resourceTriggerDefinitionUpdate),
+		Delete:      wrap(resourceTriggerDefinitionDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the Trigger Definition",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"description": {
```

Comment:
description and filter fields were the only nullable fields that I could see when looking at the Trigger Def type.  I tried setting other fields to null and API would return an error so just made those required and the nullable ones optional.